src/awkward/highlevel.py

```diff
@@ -1387,12 +1387,62 @@ def _repr(self, limit_cols):
 
         return f"<{pytype}{valuestr}{axisstr} type={typestr}>"
 
+    def _show_rows(
+        self,
+        limit_rows=20,
+        limit_cols=80,
+        type=False,
+        named_axis=False,
+        nbytes=False,
+        backend=False,
+        *,
+        formatter=None,
+        precision=3,
+    ) -> list[str]:
+        rows = []
+        formatter_impl = Formatter(formatter, precision=precision)
+
+        valuestr = prettyprint_valuestr(
+            self, limit_rows, limit_cols, formatter=formatter_impl
+        )
+        rows.append(valuestr)
+
+        if type:
+            out_io = io.StringIO()
+            out_io.write("type: ")
+            self.type.show(stream=out_io)
+            rows.append(out_io.getvalue())
+
+        # other info
+        if named_axis and self.named_axis:
+            out_io = io.StringIO()
+            out_io.write("axes: ")
+            out_io.write(
+                _prettify_named_axes(self.named_axis, delimiter=", ", maxlen=None)
+            )
+            rows.append(out_io.getvalue())
+        if nbytes:
+            out_io = io.StringIO()
+            out_io.write(f"nbytes: {bytes_repr(self.nbytes)}")
+            rows.append(out_io.getvalue())
+        if backend:
+            out_io = io.StringIO()
+            out_io.write(f"backend: {self.layout.backend.name}")
+            rows.append(out_io.getvalue())
```

Comment:
I understand why `type` uses a `io.StringIO`, because that's the way to get a string out of its `show` method (unless passing `stream=None` does it, but if so, it's because `stream=None` is implemented as "return a string, rather than modifying a stream").

But the named axes, nbytes, and backend don't need to, and it seems wasteful to create a `io.StringIO`, write one line to it, and convert that to a string. Not necessarily wasteful in computational resources (I don't know how `io.StringIO` is implemented; it might be fine), but it's a roundabout dance that brings in extra concepts without an obvious benefit.

Is it for symmetry with the `type` case? Symmetry arguments can make sense, but this one seems like a bit of a stretch.

Reply:
That's right, I can change that, the roundtrip is unnecessary. :+1:

Reply:
all comments addressed